### PR TITLE
Feature/month congratulations

### PR DESCRIPTION
**Pull request by 1MPULSEONE**

### CODE REVIEW

File: tvApp/src/main/java/band/effective/office/tv/domain/use_cases/EventStoryDataUseCase.kt
Comment (UserNameMax) [inline tvApp/src/main/java/band/effective/office/tv/domain/use_cases/EventStoryDataUseCase.kt:1]: We need to think about making it easier to add a new congratulations

**Review by UserNameMax — APPROVED** [review]

File: tvApp/build.gradle.kts [hunk 1]
```diff
@@ -54,12 +54,12 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_17
-        targetCompatibility = JavaVersion.VERSION_17
+        sourceCompatibility = JavaVersion.VERSION_11
```
Comment (gull192) [inline tvApp/build.gradle.kts:57]: why did you omit the java version?

File: tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt [hunk 2]
```diff
@@ -14,47 +11,57 @@ class EmployeeInfoEntity(
...
                 BirthdayUI(
                     employee.firstName,
                     employee.photoUrl,
                 )
-            )
-        }
-        if (employee.startDate.isNotBlank() && isCelebrationToday(employee.startDate)) {
-            resultList.add(
-                AnniversaryUI(
+            } else null,
```
Comment (gull192) [inline tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: you can use 'MutableList' and not add null. Also, why create nested arrays and then create a linear one from them. This is solved by a 'mutableList' and 'when'

File: tvApp/src/main/java/band/effective/office/tv/network/notion/EmployeeInfoRemoteDataSourceImpl.kt [hunk 3]
```diff
@@ -14,7 +14,7 @@ import java.net.UnknownHostException
 import java.util.concurrent.TimeoutException
 import javax.inject.Inject
 import javax.inject.Singleton
-
+@Deprecated("Let's use WorkTogether")
```
Comment (gull192) [inline tvApp/src/main/java/band/effective/office/tv/network/notion/EmployeeInfoRemoteDataSourceImpl.kt:17]: )))

File: tvApp/src/main/java/band/effective/office/tv/repository/workTogether/WorkTogetherImpl.kt [hunk 4]
```diff
@@ -36,12 +37,13 @@ class WorkTogetherImpl @Inject constructor(private val notionClient: NotionClien
             nextBDay = getDateFromProp("Next B-DAY"),
             workEmail = getStringFromProp("Effective Email"),
             personalEmail = getStringFromProp("Personal Email") ?: "",
-            duolingo = getStringFromProp("Профиль Duolingo")
+            duolingo = getStringFromProp("Профиль Duolingo"),
```
Comment (gull192) [inline tvApp/src/main/java/band/effective/office/tv/repository/workTogether/WorkTogetherImpl.kt:40]: It is advisable to put it in some constant

File: tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt (hunk 2 above)
Comment (UserNameMax) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: Null will filter later

Comment (gull192) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: We need to simplify this. Time for brainstorming

Comment (UserNameMax) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: I think create MutableList for mapping is not best practics

**Review by gull192 — APPROVED** [review]
If there is a desire, then you can fix the comments)

File: tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt (hunk 2 above)
Comment (gull192) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: why? The code will become more readable and there will be no dancing with tambourines with `flatten()` As a temporary solution, fits better than this

Comment (gull192) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: > I think create MutableList for mapping is not best practics
why? The code will become more readable and there will be no dancing with tambourines with flatten() As a temporary solution, fits better than this

Comment (UserNameMax) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: I think it's readable. I define congratulations for all employee and then combine them into a common list. Muttable list not match functional style

Comment (1MPULSEONE) [reply tvApp/src/main/java/band/effective/office/tv/domain/model/notion/EmployeeInfoEntity.kt:22]: If I remember correctly, he used to use the `MutableList  `implementation before, but there was a problem with the order of the checks there. I suggest leaving it that way, but I'll keep a note for potential refactoring.

**Pull request merged**